src/test/java/org/ossgang/commons/observables/operators/BlockingOperatorsTest.java

```diff
@@ -0,0 +1,78 @@
+import static org.ossgang.commons.observables.Observables.dispatcher;
+
+public class BlockingOperatorsTest {
+    public static final Duration TIMEOUT = Duration.ofSeconds(10);
+    public static final Duration REACHABLE_TIMEOUT = Duration.ofMillis(100);
+    private Dispatcher<String> observable = dispatcher("initial");
+
+    @Test
+    public void awaitNextValue_shouldIgnoreExceptionsAndReturnNextValue() {
+        CompletableFuture<String> nextValue = supplyAsync(() -> observable.awaitNextValue(TIMEOUT));
+        waitForCompletableFuture();
+        observable.dispatchException(new Exception("some exception"));
+        observable.dispatchException(new Exception("another exception"));
+        assertThat(nextValue.isDone()).isFalse();
+        String dispatchedUpdate = "UPDATE";
+        observable.dispatchValue(dispatchedUpdate);
+        assertThat(nextValue.join()).isEqualTo(dispatchedUpdate);
+    }
+
+    @Test
+    public void awaitNext_nextUpdateIsException_shouldReturnException() {
+        CompletableFuture<Maybe<String>> nextValue = supplyAsync(() -> observable.awaitNext(TIMEOUT));
+        waitForCompletableFuture();
+        Exception dispatchedException = new Exception("some exception");
+        observable.dispatchException(dispatchedException);
+        assertThat(nextValue.join().exception()).isEqualTo(dispatchedException);
+    }
+
+    @Test
+    public void awaitNext_nextUpdateIsValue_shouldReturnValue() {
+        CompletableFuture<Maybe<String>> nextValue = supplyAsync(() -> observable.awaitNext(TIMEOUT));
+        waitForCompletableFuture();
+        String dispatchedUpdate = "UPDATE";
+        observable.dispatchValue(dispatchedUpdate);
+        assertThat(nextValue.join().value()).isEqualTo(dispatchedUpdate);
+    }
+
+    @Test
+    public void awaitNextValue_noValueUpdateWithinTimeout_shouldThrow() {
+        CompletableFuture<String> nextValue = supplyAsync(() -> observable.awaitNextValue(REACHABLE_TIMEOUT));
+        waitForCompletableFuture();
+        observable.dispatchException(new Exception("some exception"));
+        observable.dispatchException(new Exception("another exception"));
+        assertThat(nextValue.isDone()).isFalse();
+        assertThatExceptionOfType(CompletionException.class).isThrownBy(nextValue::join)
+                .withCauseExactlyInstanceOf(AwaitTimeoutException.class)
+                .withMessageContaining("Timeout exceeded");
+    }
+
+    @Test
+    public void awaitNext_noUpdateWithinTimeout_shouldThrow() {
+        CompletableFuture<Maybe<String>> nextValue = supplyAsync(() -> observable.awaitNext(REACHABLE_TIMEOUT));
+        waitForCompletableFuture();
+        assertThatExceptionOfType(CompletionException.class).isThrownBy(nextValue::join)
+                .withCauseExactlyInstanceOf(AwaitTimeoutException.class)
+                .withMessageContaining("Timeout exceeded");
+    }
+
+    private void waitForCompletableFuture() {
```

Comment:
Unfortunately, I could not think of a better way to synchronize "until the subscribe happened on the completeablefuture thread" (apart from some dark reflection magic on the Observable).

Any suggestions would be appreciated.

Reply:
that's a tricky question :smile: 
Unless we introduce a, almost legal, backdoor like subscription count on the observables I'm afraid we cannot do much apart from tweaking the timeouts...

Reply:
Thanks, that's also what I thought ...
So I would say, let's see if this is going make us any problems (like unstable builds) ... if it does, we can introduce a backdoor method on one of the observable implementations ... if not, I would avoid polluting production code for the sake of easier testing ;)